vignettes/deepdive.Rmd

```diff
@@ -0,0 +1,404 @@
+
+This example is slightly different from previous examples. You will need to use `xportr_type()` to coerce R Date variables and others types to character or numeric. Only then can you use `xportr_format()` to apply the format column to the dataset.
+
+```{r, echo = TRUE}
+adsl_fmt <- adsl %>%
+  xportr_type(metadata = var_spec, domain = "ADSL", verbose = "warn") %>%
+  xportr_format(metadata = var_spec, domain = "ADSL", verbose = "warn")
+```
+
+Success! We have taken the metadata formats and applied them to the dataset. Please inspect variables like `TRTSDT` or `DISONSDT` to see the `DATE9.` format being applied.
+
+```{r, max.height='300px', attr.output='.numberLines', echo = FALSE}
+str(adsl_fmt)
+```
+
+At the time of `{xportr} v0.3.0` we have not impleneted any warnings or error messaging for this function. However, `xportr_write()` through `xpt_validate()` will check that formats have been applied for Date-Time variables.
+
+## `xportr_write()`
+
+Finally, we want to write out an `xpt` dataset with all our metadata applied. 
+
+We will make use of `xportr_metadata()` to reduce repetitive metadata and domain specifications. We will use default option for verbose, which is just `message` and so not set anything for `verbose`. In `xportr_write()` we will specify the path, which will just be our current working directory, set the dataset label and toggle the `strict_checks` to be `FALSE`.  
+
+```{r, echo = TRUE, error = TRUE}
+adsl %>%
+  xportr_metadata(var_spec, "ADSL") %>%
+  xportr_type() %>%
+  xportr_length() %>%
+  xportr_label() %>%
+  xportr_order() %>%
+  xportr_format() %>%
+  xportr_write(path = "adsl.xpt", label = "Subject-Level Analysis Dataset", strict_checks = FALSE)
+```
+
+Success! We have applied types, lengths, labels, ordering and formats to our dataset. Note the messages written out to the console. Remember the `TRTDUR` and `DCREASCD` and how these are not present in the metadata, but in the dataset. This impacts the messaging for lengths and labels where `{xportr}` is printing out some feedback to us on the two issues. 5 types are coerced, as well as 36 variables re-ordered. Note that `strict_checks` was set to `FALSE`.
+
+The next two examples showcase the `strict_checks = TRUE` option in `xportr_write()` where we will look at formats and labels.
+
+```{r, echo = TRUE, error = TRUE}
+adsl %>%
+  xportr_write(path = "adsl.xpt", label = "Subject-Level Analysis Dataset", strict_checks = TRUE)
+```
+
+
+As there at several `---DT` type variables, `xportr_write()` detects the lack of formats being applied. To correct this remember you can use `xportr_type()` and `xportr_format()` to apply formats to your xpt dataset.
+
```

Comment:
The message for format is due to DATE9., it is not recognized as a SAS format for the moment.